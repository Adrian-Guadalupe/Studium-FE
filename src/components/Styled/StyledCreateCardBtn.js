import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { Button, Icon } from 'antd';

const StyledCreateCardBtn = ({
  text,
  type = 'primary',
  size = 'default',
  icon,
  shape,
  loading,
  block,
  visable,
  ...props
}) => {
  if (visable) {
    return (
      <StyledAntdButton
        type={type}
        size={size}
        shape={shape}
        loading={loading && 'loading'}
        block={block && 'block'}
        {...props}
      >
        {
          <Icon
            type={icon}
            style={{
              fontSize: '59px',
              color: 'black',
              lineHeight: '10px',
              fontWeight: 900,
            }}
          />
        }
        {text}
      </StyledAntdButton>
    );
  } else {
    return <></>;
  }
};

StyledCreateCardBtn.propTypes = {
  text: PropTypes.string,
  type: PropTypes.oneOf(['primary']),
  size: PropTypes.oneOf(['large', 'default', 'small']),
  icon: PropTypes.string,
  shape: PropTypes.oneOf(['circle', 'round']),
  spin: PropTypes.bool,
  loading: PropTypes.bool,
  block: PropTypes.bool,
  left: PropTypes.string,
  bottom: PropTypes.string,
  padding: PropTypes.string,
  border: PropTypes.string,
  onClick: PropTypes.func.isRequired,
};

const StyledAntdButton = styled(Button)`
  && {
    position: absolute;
    left: ${props => props.left || '50%'};
    bottom: ${props => props.bottom || '25px'};
    width: 99px;
    height: 99px;
    padding: ${props => props.padding || '0'};
    background: ${props => props.backgroundColor || props.theme.lightGray};
    border: ${props => props.border || '10px solid white'};
    display: flex;
    justify-content: center;
    align-items: center;
    box-sizing: border-box;
    transform: translate(-50%);
  }
`;
export default StyledCreateCardBtn;
