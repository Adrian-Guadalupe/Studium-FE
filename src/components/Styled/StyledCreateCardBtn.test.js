import React from 'react';
import renderer from 'react-test-renderer';

import StyledCreateCardBtn from './StyledCreateCardBtn';

test('StyledCreateCardBtn render correctly', () => {
    const tree = renderer.create(<StyledCreateCardBtn />).toJSON();
    expect(tree).toMatchSnapshot();
});