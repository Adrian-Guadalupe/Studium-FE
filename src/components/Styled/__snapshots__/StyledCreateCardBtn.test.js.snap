// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledCreateCardBtn render correctly 1`] = `
.c0.c0 {
  position: absolute;
  width: 99px;
  height: 99px;
  left: 139px;
  top: 688px;
  padding: 10px;
  background: #d7d7d7;
  border-style: none;
  box-sizing: border-box;
}

<button
  className="ant-btn c0 ant-btn-primary"
  onClick={[Function]}
  type="button"
>
  <i
    className="anticon"
    style={
      Object {
        "color": "black",
        "fontSize": "70px",
        "lineHeight": "10px",
      }
    }
  />
</button>
`;
